### PR TITLE
Revise Kafka Streams docs

### PR DESCRIPTION
**Pull request by sabbyanandan**
In this proposal, I attempted to bring consistency to KStream docs similar to how Rabbit and Kafka sections are structured. 

- Revise the top-level sections to be consistent w/ other binder docs 
- Change the ordering for better continuity 
- Rephrase related section docs
- Fix typos

Resolves spring-cloud/spring-cloud-stream-binder-kafka#314

### CODE REVIEW

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc [hunk 1]
```diff
@@ -1,29 +1,49 @@
...
+[source,xml]
+----
+<dependency>
+  <groupId>org.springframework.cloud</groupId>
+  <artifactId>spring-cloud-stream-binder-kafka-streams</artifactId>
+</dependency>
+----
 
-Here are the maven coordinates for the Spring Cloud Stream Kafka Streams binder artifact.
+Alternatively, you can also use the Spring Cloud Stream RabbitMQ Starter.
```
Comment (sobychacko) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc]: Why the mention of RabbitMQ starter here?

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc [hunk 2]
```diff
@@ -1,29 +1,49 @@
...
 
 [source,xml]
 ----
 <dependency>
   <groupId>org.springframework.cloud</groupId>
-  <artifactId>spring-cloud-stream-binder-kafka-streams</artifactId>
+  <artifactId>spring-cloud-starter-stream-kafka-streams</artifactId>
```
Comment (sobychacko) [inline spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc]: We don't provide a starter artifact for kafka streams binder yet?

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc (hunk 1 above)
Comment (sabbyanandan) [reply spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc]: Good catch, I was reusing the text from Rabbit binder, but I forgot to change it. 🤦‍♂️ - I'll fix it.

File: spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc (hunk 2 above)
Comment (sabbyanandan) [reply spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/kafka-streams.adoc]: Fixing it.

**Pull request merged**